functions_for_gyre.py

```diff
@@ -41,50 +42,100 @@ def calc_scanning_range(gyre_file_path, npg_min=-50, npg_max=-1, l=1, m=1, omega
+        df = pandas.read_csv(os.path.expandvars('$CONDA_PREFIX/lib/python3.7/site-packages/mamsie_py/lambda.csv'), sep=',')
+        
+        # will add extra functionality to calculate the bounds explicitly, making use of GYRE
+
+        # Select nu (spin parameter) and lambda column when values in l and m column correspond to requested values
+        ###### SHOULD BE CHANGED TO PARAMETER 'K' ---> needs adjustment in lambda.csv - JVB.
         NuLambda = df.loc[(df['l'] == l) & (df['m'] == m)][['nu', 'Lambda']]
 
+        # Generate numpy array from pandas dataframe series
         nu = NuLambda['nu'].to_numpy()
         Lambda = NuLambda['Lambda'].to_numpy()
 
+        # Generate difference between pulsation frequency and asymptotic value (in co-rotating frame) in units of c/d
         diff_max = nu/(2.*omega_rot) - P0*n_max_used/np.sqrt(Lambda)
         diff_min = nu/(2.*omega_rot) - P0*n_min_used/np.sqrt(Lambda)
+        # Obtain index of minimal difference/distance
         index_max = np.where(abs(diff_max) == np.min(abs(diff_max)))[0]
         index_min = np.where(abs(diff_min) == np.min(abs(diff_min)))[0]
-        f_min = (np.sqrt(Lambda[index_max]) / (P0*n_max_used) + m*omega_rot)[0]
-        f_max = (np.sqrt(Lambda[index_min]) / (P0*n_min_used) + m*omega_rot)[0]
+        # Calculate the rotationally shifted frequency (TAR approximation)
+        ### in the inertial frame
+        if frame == 'inertial':
+            f_min = (np.sqrt(Lambda[index_max]) / (P0*n_max_used) + m*omega_rot)[0]
+            f_max = (np.sqrt(Lambda[index_min]) / (P0*n_min_used) + m*omega_rot)[0]
+        ### in the co-rotating frame
+        else:
+            f_min = (np.sqrt(Lambda[index_max]) / (P0*n_max_used))[0]
+            f_max = (np.sqrt(Lambda[index_min]) / (P0*n_min_used))[0]
     return f_min, f_max
 
+################################################################################
+################################################################################
+# Function written by Jordan Van Beeck
+################################################################################
+
+def calculate_k(l,m,rossby):
+  """
+    Compute the mode classification parameter for gravity or Rossby modes from the corresponding azimuthal order (m) and spherical degree (l). 
+    Raises an error when l is smaller than m.
+    ------- Parameters -------
+    rossby: boolean
+        parameter that needs to be set to True if Rossby mode k is calculated
+    l, m: integer
+        degree (l) and azimuthal order (m) of the modes
+
+    ------- Returns -------
+    k: integer 
+        mode classification parameter of the pulsation mode
+  """
+  if not rossby:
+    # g-mode k
+    if abs(l) >= abs(m):
+      k = l - abs(m) # Lee & Saio (1997)
+      return k
+    else:
+      raise Exception('l is smaller than m, please revise your script/logic. The corresponding values were: (l,m) = ({lval},{mval})'.format(lval=l,mval=m))
```

Comment:
use f-strings for formatting, so line could be
raise Exception(f'l is smaller than m, please revise your script/logic. The corresponding values were: (l,m) = ({l},{m})

Reply:
`raise Exception(f'l is smaller than m, please revise your script/logic. The corresponding values were: (l,m) = ({l},{m})`